common/transport/amqp/src/amqp.ts

```diff
@@ -132,6 +88,185 @@ export class Amqp {
+          '*': () => this._fsm.deferUntilTransition('connected')
+        },
+        connected: {
+          connect: (callback) => callback(null, new results.Connected()),
+          disconnect: (disconnectCallback) => {
+            this._fsm.transition('disconnecting');
+            this._disconnect((err) => {
+              this._fsm.transition('disconnected');
+              disconnectCallback(err);
+            });
+          },
+          initializeCBS: (callback) => {
+            this._cbs = new ClaimsBasedSecurityAgent(this._amqp);
+            this._cbs.attach(callback);
+          },
+          putToken: (audience, token, callback) => {
+            this._cbs.putToken(audience, token, callback);
+          },
+          send: (message: Message, endpoint: string, to: string, done: GenericAmqpBaseCallback): void => {
+            /*Codes_SRS_NODE_COMMON_AMQP_16_006: [The `send` method shall construct an AMQP message using information supplied by the caller, as follows:
+            The ‘to’ field of the message should be set to the ‘to’ argument.
+            The ‘body’ of the message should be built using the message argument.] */
+
+            let amqpMessage = AmqpMessage.fromMessage(message);
+            if (to !== undefined) {
+              amqpMessage.properties.to = to;
+            }
+
+            if (!this._senders[endpoint]) {
+              this._fsm.handle('attachSenderLink', endpoint, null, (err) => {
+                if (err) {
+                  done(err);
+                } else {
+                  (this._senders[endpoint] as SenderLink).send(amqpMessage, done);
+                }
+              });
+            } else {
+              (this._senders[endpoint] as SenderLink).send(amqpMessage, done);
+            }
+
+          },
+          getReceiver: (endpoint: string, done: GenericAmqpBaseCallback): void => {
+            /*Codes_SRS_NODE_COMMON_AMQP_16_010: [If a receiver for this endpoint doesn’t exist, the getReceiver method should create a new AmqpReceiver object and then call the done() method with the object that was just created as an argument.] */
+            if (!this._receivers[endpoint]) {
+              this._fsm.handle('attachReceiverLink', endpoint, null, (err) => {
+                if (err) {
+                  done(err);
+                } else {
+                  done(null, this._receivers[endpoint]);
+                }
+              });
+            } else {
+              /*Codes_SRS_NODE_COMMON_AMQP_16_009: [If a receiver for this endpoint has already been created, the getReceiver method should call the done() method with the existing instance as an argument.] */
+              this._safeCallback(done, null, this._receivers[endpoint]);
+            }
+          },
+          attachReceiverLink: (endpoint: string, linkOptions: any, done: GenericAmqpBaseCallback): void => {
+            let receiverFsm = new ReceiverLink(endpoint, linkOptions, this._amqp);
```

Comment:
>Fsm [](start = 24, length = 3)

You may call it an FSM, and it may have a private member that is an fsm, but that really doesn't seem to make this an fsm.